src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -260,26 +269,48 @@ function show() {
   }
 }
 
+function waitForPopoverFocus() {
+  return new Promise((resolve) => {
+    requestAnimationFrame(() => {
+      requestAnimationFrame(() => {
+        requestAnimationFrame(resolve);
+      });
+    });
+  });
+}
+
 async function afterShow() {
   setOverflowHeight();
 
   emit('after-open');
 
   if (props.isSearchEnabled) {
-    await new Promise((resolve) => {
-      requestAnimationFrame(() => {
-        requestAnimationFrame(() => {
-          requestAnimationFrame(resolve);
-        });
-      });
-    });
-
+    await waitForPopoverFocus();
```

Comment:
We can move all three `waitForPopoverFocus` before the `if` on the line 287 as it is called in every condition.

Reply:
corrected thanks